lib/src/package_config.dart

```diff
@@ -274,3 +276,24 @@ String extractLanguageVersion(VersionConstraint c) {
   }
   return '${minVersion.major}.${minVersion.minor}';
 }
+
+/// Parses the (now obsolete) .packages file.
+Map<String, Uri> parsePackagesFile(List<int> bytes, Uri base) {
+  final result = <String, Uri>{};
+  for (var line in utf8.decode(bytes).split('\n')) {
+    line = line.trim();
+    final commentLocation = line.indexOf('#');
+    if (commentLocation != -1) line = line.substring(0, commentLocation);
+    if (line.isEmpty) continue;
+    final colonLocation = line.indexOf(':');
+    if (colonLocation == -1) {
+      throw FormatException('Malformed line in .packages');
+    }
+    final packageName = line.substring(0, colonLocation);
+    if (result.containsKey(packageName)) {
+      throw FormatException('.packages contains $packageName twice');
+    }
+    result[packageName] = base.resolve(line.substring(colonLocation + 1));
```

Comment:
Should we catch `FomatException` from `base.resolve` and throw it as a `FormatException` with a nicer message?